pymc3/backends/base.py

```diff
@@ -552,17 +554,26 @@ def merge_traces(mtraces):
 
     Raises
     ------
-    A ValueError is raised if any traces have overlapping chain numbers.
+    A ValueError is raised if any traces have overlapping chain numbers,
+    or if chains are of different lengths.
 
     Returns
     -------
     A MultiTrace instance with merged chains
     """
+    if len(mtraces) == 0:
+        raise ValueError("Cannot merge an empty set of traces.")
     base_mtrace = mtraces[0]
+    chain_len = len(base_mtrace)
+    # check base trace
+    if any((len(st) != chain_len for _, st in base_mtrace._straces.items())):
```

Comment:
Yeah that's more readable. Does this need some kind of `#` flag so pylint ignores the long line?

Reply:
also, you can get rid of the internal parens:
```suggestion
    if any(len(st) != chain_len for _, st in base_mtrace._straces.items()):  # pylint: disable=line-too-long
```